src/main/scala/devsearch/features/QueryRecognizer.scala

```diff
@@ -0,0 +1,37 @@
+package devsearch.features
+
+import devsearch.parsers.Languages
+
+object QueryRecognizer {
+  def apply(query: String): Option[CodeFile] = {
+    scala.util.Try {
+      Languages.supportedLanguages().toList.map { language =>
+        // Some parsers (Go) need a valid file name to parse correctly, so we
+        // generate one for the query.
+        val generatedFileName = "file" + (Languages.extension(language) match {
+          case Some(extension) => "." + extension
+          case None => ""
+        })
+
+        CodeFile(
+          language,
+          CodeFileLocation("dummy_user", "dummy_repo", generatedFileName),
+          query
+        )
+      }.map { codeFile =>
+        // Scala parser is more permissive, so we reduce its score for the comparison.
+        val weight: Double = codeFile.language match {
+          case Languages.Scala => 0.7
```

Comment:
This is rather arbitrary... The problem with the query parser is that it can probably parse most of Java and while doing so it introduces synthetic ASTs, so we get more features than with the java parser.